### ISSUE TITLE
Add mouse forward/back button support

### ISSUE THREAD

**Issue by twisterghost** [brave/browser-laptop#938]
Plenty of mice these days have side buttons for forward/back while browsing the web. I'd like to see support for this in Brave.

I did some poking around at it for a while, but unfortunately came up with no fruitful results. There doesn't seem to be a reasonable way to detect special mouse button presses (or anything beyond left/right/middle click) within normal client JS-land, thus my approach of putting a listener in the main component did not work out.

I tried to do some research into electron to see what kind of native mouse event support there was, but nothing I could find was too useful.

Long story short, no great avenue seems immediately available. If anyone has knowledge in this area to shed some light, I'd be happy to work on a pull request proposal for this.

**Comment by bbondy** [comment]
Try checking `e.button` in a click event, I bet the value changes for those buttons.

**Comment by twisterghost** [comment]
The problem is the buttons don't even seem to trigger the event, otherwise yeah, that would be simple enough. Seems like natively, mouse events only trigger from mouse button 1, 2 or 3, but not 4 or 5 which seem to be generally supported as forward/back

**Comment by bbondy** [comment]
I think this might be it:
https://github.com/atom/electron/blob/master/docs/api/browser-window.md#event-app-command-windows

```
Event: 'app-command' Windows

Emitted when an App Command is invoked. These are typically related to keyboard media keys or browser commands, as well as the "Back" button built into some mice on Windows.

someWindow.on('app-command', function(e, cmd) {
  // Navigate the window back when the user hits their mouse back button
  if (cmd === 'browser-backward' && someWindow.webContents.canGoBack()) {
    someWindow.webContents.goBack();
  }
});
```

**Comment by bbondy** [comment]
try adding it inside js/stores/appStore.js

**Comment by twisterghost** [comment]
Will do

**Comment by twisterghost** [comment]
This is going to require working on `mainWindow` in `appStore.js` but also getting a handle of the current active frame. Do you have any suggestions on the best way to get those two values in one place? I can think of some ways that may be counter-intuitive to the design so far, like sending an action from appStore to windowStore or handling it in a component, but that seems silly.

**Comment by twisterghost** [comment]
PR: https://github.com/brave/browser-laptop/pull/944

**Comment by hasanhuseyincakir** [comment]
Can this be added to GUI at Right Click dropdown menu?

**Comment by rmathus4** [comment]
Agreed - I'd love to see this with the right-click mouse button for back/forward.  I often find myself hitting Reload rather than the expected option of Back.

**Comment by bbondy** [comment]
@hasanhuseyincakir @rmathus4  do you mean just a contedxt menu item for Back and Forward?
If so pls just file a new issue for this. Would be straight forward to add.

**Comment by bbondy** [comment]
Actually I posted here:
https://github.com/brave/browser-laptop/issues/1240

So only post a new issue if it's something different than that.

**Comment by hasanhuseyincakir** [comment]
I prefer Firefox Style Back - Forward buttons but Text would do the same. 
![fff](https://cloud.githubusercontent.com/assets/757853/14364381/4cb7cdec-fd11-11e5-84da-c8c33530fb4c.png)

**Comment by bbondy** [comment]
This was added for 0.9, we can iterate to icons later if we want to.
